src/snake.rs

```diff
@@ -1,39 +1,77 @@
-use crate::location::*;
-
 use bevy::prelude::*;
+use bevy::core::FixedTimestep;
+use bevy::input::keyboard::KeyboardInput;
+use bevy::sprite::Anchor;
+use crate::location::*;
+use crate::direction::{Direction};
 
 pub struct Plugin;
 
 impl bevy::prelude::Plugin for Plugin {
     fn build(&self, app: &mut App) {
         app.add_startup_system(setup)
-            .add_system(list_snakes);
+            .add_system_set(
+                SystemSet::new()
+                    .with_run_criteria(FixedTimestep::step(1.0))
+                    .with_system(move_snakes)
+            )
+            .add_system(update_directions)
+            .add_system(render_heads);
     }
 }
 
 fn setup(mut commands: Commands) {
     commands.spawn()
         .insert(Head)
-        .insert(Location::default())
-        .insert(Moving::default());
+        .insert(Location::new())
+        .insert(Moving::default())
+        .insert_bundle(SpriteBundle {
+            sprite: Sprite {
+                anchor: Anchor::TopLeft,
+                custom_size: Some(Vec2::new(80.0, 80.0)),
+                color: Color::rgb(0.06, 0.46, 0.06),
+                ..default()
+            },
+            ..default()
+        });
 }
 
 #[derive(Component)]
 struct Head;
 
-#[derive(Default, Debug)]
-enum Direction {
-    #[default]
-    Up, Down, Left, Right,
-}
-
 #[derive(Component, Default, Debug)]
 struct Moving {
-    dir: Direction,
+    last_dir: Direction,
```

Comment:
I haven't thought this through fully, but I suspect we could ignore "backwards" inputs without needing to remember this.